testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -581,3 +584,289 @@ def test_segmentgroup_repr(self):
     def test_segmentgroup_str(self):
         sg = self.u.segments[:10]
         assert_(str(sg) == '<SegmentGroup [<Segment 4AKE>]>')
+
+
+
```

Comment:
Impressive battery of tests!

The only thing that seems to be missing is tests for the operator versions, `-`, `&`, `|`, `^`.